### ISSUE TITLE
Add automatic running of a command in the REPL on buffer-load

### ISSUE THREAD

**Issue by zendevil** [clojure-emacs/cider#2818]
I have a repl command ```(go)``` that restarts a server, but whenever I make a change in the server code, I have to first do ```C-c C-k (C-e)``` and then enter the command ```(go)```. I want to automate this process, so that whenever I do ```C-c C-k (C-e)```, the server is automatically restarted, i.e., ```(go)``` is called automatically. Currently, nothing in the docs tell me how to do this.

**Comment by dpsutton** [comment]
https://docs.cider.mx/cider/usage/misc_features.html#_reloading_code

This might help

```
(setq cider-ns-refresh-before-fn "user/stop-system!"
      cider-ns-refresh-after-fn "user/start-system!")
```

**Comment by zendevil** [comment]
Adding 

```
(setq cider-ns-refresh-before-fn "user/go"
      cider-ns-refresh-after-fn "user/go")
```
in .spacemacs hasn't worked for me. It seems like ```go``` is still not called on C-c C-k. I don't see the restart prints I usually do when doing (go) in the repl.

**Comment by dpsutton** [comment]
> You can define Clojure functions to be called before reloading, and after a successful reload, when using cider-ns-refresh:

You are calling cider eval buffer. If you use cider-ns-refresh it should invoke these post refresh functions. Is this workflow amenable to you or must it trigger on `cider-load-buffer`. If not, write your own function that invokes `cider-load-buffer` and then calls a function. Alternatively, `cider-load-buffer` takes a callback which is normally `cider-load-file-handler`. And you can make one of these with an optional `done-handler` which could call your function.

**Comment by zendevil** [comment]
No, in fact, doing cider-ns-refresh doesn't trigger go, and in fact breaks the repl, i.e., I can't even run go manually after cider-ns-refresh. 
@dpsutton How do I invoke cider-load-buffer specifically? Would you please give a snippet?

**Comment by dpsutton** [comment]
the done-handler is if you want to get fancy and you can look into that. fundamentally you want something like

```lisp
(defun custom/load-buffer ()
  (interactive)
  (cider-load-buffer)
  (cider-interactive-eval "(+ 1 1)"))

(defun custom/load-buffer-2 ()
  (interactive)
  (cider-load-buffer (current-buffer)
                     (cider-load-file-handler nil
                                              (lambda (_buffer)
                                                (cider-interactive-eval "(+ 1 1)")))))
```

**Comment by zendevil** [comment]
Thanks for the code. I tried the first example. The only thing is that I also have another buffer to load, which doesn't change regardless of what the current buffer is, before calling (go), i.e., 

(cider-load-buffer) 
(cider-load-buffer "some/namespace")

But when I use the second line, the eval step stops working. How to reference another buffer correctly (not the current one) when passing into cider-load-buffer?

**Comment by dpsutton** [comment]
there's also a `get-buffer-create` that can create a buffer if it doesn't exist.
